test/parse-test.js

```diff
@@ -107,71 +120,121 @@ describe("dateFormat.parse", function() {
       verifyDate(date, { year: 2015, month: 8 });
     });
 
-    it("should match all the date parts", function() {
-      var date = dateFormat.parse("dd", "21");
-      verifyDate(date, { day: 21 });
+    describe("should match all the date parts", function() {
+      // Test cases without timezone are invalid as the result varies depends on
+      // OS timezone and the hour the test cases are executed.  
+      // These use cases should not be supported in the first place.  Using 
+      // them will ensure all your time filled with sadness.
+      xit("works with dd", function() {
```

Comment:
I think these tests are still valid. My main aim is to make sure the code does something predictable with limited data. If no timezone is specified, then we should compare against a local time. So `dateFormat.parse("dd","21")` should return the same as `(new Date()).setDate("22")` - what do you think?

Reply:
I looked for hours and don't see a way around it.  There is no way to get correct consistent result.  Best just not to support it or let people know use at their own risk.

Reply:
Fair enough. I'll merge and see if I can get them to work - if I can't then I'll remove the tests and put something in the docs about it.